src/Utilities/cachingoptimizer.jl

```diff
@@ -34,19 +34,24 @@ mutable struct CachingOptimizer{MT<:MOI.ModelLike} <: MOI.AbstractOptimizer
     # optimizer indices.
 end
 
-CachingOptimizer(model_cache::MOI.ModelLike, mode::CachingOptimizerMode) = CachingOptimizer(model_cache, nothing, NoOptimizer, mode, IndexMap(), IndexMap())
+function CachingOptimizer(model_cache::MOI.ModelLike, mode::CachingOptimizerMode)
+    CachingOptimizer{MOI.AbstractOptimizer, typeof(model_cache)}(nothing, model_cache, NoOptimizer, mode, IndexMap(), IndexMap())
+end
 
 """
     CachingOptimizer(model_cache::MOI.ModelLike, optimizer::AbstractOptimizer)
 
 Creates an `CachingOptimizer` in `Automatic` mode, with the optimizer `optimizer`.
 The model_cache manager returned behaves like an `AbstractOptimizer` as long as no
-`CachingOptimizer`-specific functions (e.g. `dropoptimizer!`) are called on it.
+`CachingOptimizer`-specific functions (e.g. `resetoptimizer!`) are called on it.
+The type of the optimizer returned is `CachingOptimizer{typeof(optimizer),
+typeof(model_cache)}` so it does not support the function `dropoptimizer!`
```

Comment:
> so it does not support the function `dropoptimizer!` nor the method `resetoptimizer!.

Is this true? Can't the field be set to nothing?

Reply:
Indeed, fixed :)